src/components/Image/DetailV2/ImageContestCollectionDetails.tsx

```diff
@@ -0,0 +1,110 @@
+import { Anchor, Button, Card, Divider, Text } from '@mantine/core';
+import { CollectionItemStatus, CollectionType } from '@prisma/client';
+import { IconTournament } from '@tabler/icons-react';
+import { useImageDetailContext } from '~/components/Image/Detail/ImageDetailProvider';
+import { useImageContestCollectionDetails } from '~/components/Image/image.utils';
+import { ShareButton } from '~/components/ShareButton/ShareButton';
+
+export const ImageContestCollectionDetails = ({
+  imageId,
+  isOwner,
+}: {
+  imageId: number;
+  isOwner: boolean;
+}) => {
+  const { collectionItems } = useImageContestCollectionDetails({ id: imageId });
+  const { shareUrl } = useImageDetailContext();
+  if ((collectionItems?.length ?? 0) === 0) return null;
+
+  const displayedItems =
+    collectionItems?.filter((ci) => ci.status === CollectionItemStatus.ACCEPTED || isOwner) ?? [];
+
+  if (displayedItems.length === 0) return null;
+
+  return (
+    <Card className="flex flex-col gap-3 rounded-xl">
+      <div className="flex items-center gap-3">
+        <Text className="flex items-center gap-2 text-xl font-semibold">
+          <IconTournament />
+          <span>Contests</span>
+        </Text>
+      </div>
+      <div className="flex flex-col gap-3">
+        {collectionItems?.map((item) => {
+          if (isOwner) {
+            return (
```

Comment:
At this point I think this is better as a separate component 😛 what do you think?

Reply:
I don't see a point in that, it'd only be used in this component for one, and this component is already contained enough 🤷